### ISSUE TITLE
The nx-freestance does not work with the new deb file

### ISSUE THREAD

**Issue by danrobi11** [atlas-engineer/nyxt#1080]
[nx-freestance-handler](https://github.com/kssytsrk/nx-freestance-handler) does not work with the new deb file version 2-pre-release-5. 
Bummer cuz I wanted to promote the new nyxt release / the new nx-freestance in a few subreddit.
The nx-freestance is working fine when compiling nyxt from source.
Which makes me think that the deb version by default might not be listening to whats in the ```common.lisp``` directory. 

Is it possible to manually add a few lines of code somewhere to create a sort of work around for new users that would like to install the deb file, instead of having to compile nyxt from source in order to run the nx-freestance-handler?

Although, thank you much for the new deb file release!

P.S Possible to add a note somewhere about the nx-freestance-handler so new users could know about it? In the readme maybe? idk.

**Comment by aartaka** [comment]
Regarding the note about the nx-freestance-handler: It's already there, in [EXTENSIONS.org](https://github.com/atlas-engineer/nyxt/blob/master/documents/EXTENSIONS.org) :heavy_check_mark: 

Regarding extension compatibility: A workaround could be a good old 
``` lisp
(push #P"/path/to/nx-freestance-handler/" asdf:*central-registry*)
```

But I'm not fully aware of how either of nx-freestance-handler or Nyxt extension handling works, so @Ambrevar might be in a better position to help you :)

**Comment by danrobi11** [comment]
> It's already there, in EXTENSIONS.org heavy_check_mark

Nice!
I've tried adding ```(push #P"/home/danrobi/common-lisp/nx-freestance-handler/" asdf:*central-registry*)``` in my ```init.lisp``` and unfortunately that does not work.

**Comment by jmercouris** [comment]
@danrobi11 thank you for your interest in Nyxt!

You can try `(asdf:load-asd "/path/to/asd/file")` then you should be able to load the system with `(asdf:load-system :system-name)`.

Of course substituting the above with your correct information.

**Comment by jmercouris** [comment]
If it doesn't work in your init file, please try it in the `lisp-repl` and report any errors/messages in `list-messages`. Thank you!

**Comment by danrobi11** [comment]
@jmercouris 
With ```(asdf:load-asd "/home/danrobi/common-lisp/nx-freestance-handler")``` I get this warning in nyxt:
```
Could not load the init file: Couldn't load #P"/home/danrobi/.cache/common-lisp/sbcl-2.0.10-linux-x64/home/runner/common-lisp/asdf/build/asdf.fasl": file does not exist.
```
I've created a ```lisp.repl``` file in ```~/.config/nyxt``` with the ```(asdf:load-asd ...``` line in it and that does nothing. 

I've also created a ```list.messages``` which I've tried in ```~/.config/nyxt``` and also in ```~/common-lisp``` nothing gets written in there .

**Comment by jmercouris** [comment]
Hi @danrobi11, I'm sorry I wasn't clear enough,

`lisp-repl` and `list-messages` are commands in Nyxt. I was suggesting to try it interactively and see what happens.

As per the first line you must write `(asdf:load-asd "/home/dandrobi/common-lisp/nx-freestance-handler/nx-freestance-handler.asd")`

You are missing the path to the 'asd' file specifically. Please give it a try with this change :-)

**Comment by Ambrevar** [comment]
This line

     Could not load the init file: Couldn't load #P"/home/danrobi/.cache/common-lisp/sbcl-2.0.10-linux-x64/home/runner/common-lisp/asdf/build/asdf.fasl": file does not exist.

suggests that the ASDF configuration went stale in the .deb.

@danrobi11: Can you try putting

```
(asdf:clear-configuration)
```

first thing in your init.lisp?

**Comment by danrobi11** [comment]
Nothing is working. I do appreciate the support. Thanks! @jmercouris @Ambrevar

```list-messages``` = ```WARN - Warning: Could not load the init file.```

Im not sure about the ```lisp-repl``` command. At the prompt, it wont let me paste text from clipboard :/  Im not sure about what im suppose to add there. So I manually typed the ```(asdf:load-asd "/home/dandrobi/common-lisp/nx-freestance-handler/nx-freestance-handler.asd")``` line. 
Also tried with ```(asdf:load-system :nx-freestance-handler)``` and with ```(asdf:load-system :nx-freestance-handler.asd)```. Same result, nyxt crashes when I hit ```enter```. 

Heres my ```init.lisp```. It might help figure why the ```nx-freestance-handler``` does not work with the .deb version.

```
(asdf:clear-configuration)

(asdf:load-asd "/home/danrobi/common-lisp/nx-freestance-handler/nx-freestance-handler.asd")

(push #P"/home/danrobi/common-lisp/nx-freestance-handler/" asdf:*central-registry*)

(defvar *my-request-resource-handlers* '())

(load-after-system :nx-freestance-handler
                   (nyxt-init-file "freestance.lisp"))

(define-configuration buffer
    ((request-resource-hook
      (reduce #'hooks:add-hook
              *my-request-resource-handlers*
              :initial-value %slot-default))))
                           
```

**Comment by Ambrevar** [comment]
Can you put just the following

```
(asdf:clear-configuration)

(asdf:load-asd "/home/danrobi/common-lisp/nx-freestance-handler/nx-freestance-handler.asd")
(load-after-system :nx-freestance-handler)
```

then start Nyxt from a shell and report the output?

**Comment by danrobi11** [comment]
@Ambrevar 

I get this in nyxt with the above:

```
Could not load the init file: Error while trying to load definition for system asdf from pathname /home/runner/common-lisp/asdf/asdf.asd: Couldn't load #P"/home/runner/common-lisp/asdf/asdf.asd": file does not exist.
```
Heres the output:

```
<INFO> [03:39:15] Listening to socket "/home/danrobi/.local/share/nyxt/nyxt.socket".
Nyxt version 2-pre-release-5
<INFO> [03:39:15] Loading Lisp file "/home/danrobi/.config/nyxt/init.lisp".
WARNING:
   You are using ASDF version 3.3.4.7 from
   #P"/home/runner/common-lisp/asdf/asdf.asd" and have an older version of ASDF
   (and older than 2.27 at that) registered at
   #P"/home/runner/common-lisp/asdf/asdf.asd". Having an ASDF installed and
   registered is the normal way of configuring ASDF to upgrade itself, and
   having an old version registered is a configuration error. ASDF will ignore
   this configured system rather than downgrade itself. In the future, you may
   want to either: (a) upgrade this configured ASDF to a newer version, (b)
   install a newer ASDF and register it in front of the former in your
   configuration, or (c) uninstall or unregister this and any other old version
   of ASDF from your configuration. Note that the older ASDF might be
   registered implicitly through configuration inherited from your system
   installation, in which case you might have to specify
   :ignore-inherited-configuration in your in your
   ~/.config/common-lisp/source-registry.conf or other source-registry
   configuration file, environment variable or lisp parameter. Indeed, a likely
   offender is an obsolete version of the cl-asdf debian or ubuntu package,
   that you might want to upgrade (if a recent enough version is available) or
   else remove altogether (since most implementations ship with a recent asdf);
   if you lack the system administration rights to upgrade or remove this
   package, then you might indeed want to either install and register a more
   recent version, or use :ignore-inherited-configuration to avoid registering
   the old one. Please consult ASDF documentation and/or experts.

While evaluating the form starting at line 3, column 0
  of #P"/home/danrobi/.config/nyxt/init.lisp":
; 
; compilation unit aborted
;   caught 1 fatal ERROR condition
<WARN> [03:39:16] Warning: Could not load the init file.
```

**Comment by Ambrevar** [comment]
I think this is an issue with linux-packaging.  I'll investigate.

**Comment by aartaka** [comment]
Maybe a different issue: Nyxt extensions (tested with [nx-ace](https://github.com/atlas-engineer/nx-ace), [nx-kaomoji](https://github.com/aartaka/nx-kaomoji), and [nx-search-engines](https://github.com/aartaka/nx-search-engines)) are not loadable into the Guix-compiled version of Nyxt on Guix System.

Small details:
- In `lisp-repl`: `(asdf:load-asd #p"/path/to/the.asd")` passes fine, `(asdf:load-system ...)` passes silently, but then calling any command and referensing any variable from the package in question causes Nyxt to error with the error message of a type `WARN - Warning: Error on separate thead: Package NX-SEARCH-ENGINES does not exist. Stream: #<dynamic-extent STRING-INPUT-STREAM (unavailable) from "(nx-sear...">`.
- Putting the `asdf:load-asd` and `load-after-system` in init.lisp doesn't to anything.
- `asdf:*central-registry*` is nil when checked from `lisp-repl`.
- CL packages loaded from Guix load just fine. Example: `slynk` loads alright and associated file loaded with `load-after-system :slynk` is evaluated too.

**Comment by Ambrevar** [comment]
> `asdf:*central-registry*` is nil when checked from lisp-repl.

This is expected, this variable is deprecated anyways.

Can you try running `(asdf:clear-configuration)`, then load the extension see if that helps?

**Comment by aartaka** [comment]
> > `asdf:*central-registry*` is nil when checked from lisp-repl.
> 
> This is expected, this variable is deprecated anyways.
> 
> Can you try running `(asdf:clear-configuration)`, then load the extension see if that helps?

No, it does not help -- still the same thing: it passes fine with `load-asd` and silent with `load-system`, and then errors about package not existing when I try to reference anything in it.

**Comment by Ambrevar** [comment]
Fixed with #1366.  Feed free to reopen if there is anything else.
